pkg/kubecost/allocation.go

```diff
@@ -2104,52 +2053,25 @@ func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*Allocati
 // AggregateBy aggregates each AllocationSet in the range by the given
 // properties and options.
 func (asr *AllocationSetRange) AggregateBy(aggregateBy []string, options *AllocationAggregationOptions) error {
-	aggRange := &AllocationSetRange{allocations: []*AllocationSet{}}
-
-	asr.Lock()
-	defer asr.Unlock()
+	aggRange := &AllocationSetRange{Allocations: []*AllocationSet{}}
 
-	for _, as := range asr.allocations {
+	for _, as := range asr.Allocations {
 		err := as.AggregateBy(aggregateBy, options)
 		if err != nil {
 			return err
 		}
-		aggRange.allocations = append(aggRange.allocations, as)
+		aggRange.Allocations = append(aggRange.Allocations, as)
 	}
 
-	asr.allocations = aggRange.allocations
+	asr.Allocations = aggRange.Allocations
 
 	return nil
 }
 
 // Append appends the given AllocationSet to the end of the range. It does not
 // validate whether or not that violates window continuity.
 func (asr *AllocationSetRange) Append(that *AllocationSet) {
-	asr.Lock()
-	defer asr.Unlock()
-	asr.allocations = append(asr.allocations, that)
-}
-
-// Each invokes the given function for each AllocationSet in the range
-func (asr *AllocationSetRange) Each(f func(int, *AllocationSet)) {
-	if asr == nil {
-		return
-	}
-
-	for i, as := range asr.allocations {
-		f(i, as)
-	}
-}
-
-// Get retrieves the AllocationSet at the given index of the range.
-func (asr *AllocationSetRange) Get(i int) (*AllocationSet, error) {
-	if i < 0 || i >= len(asr.allocations) {
-		return nil, fmt.Errorf("AllocationSetRange: index out of range: %d", i)
-	}
-
-	asr.RLock()
-	defer asr.RUnlock()
-	return asr.allocations[i], nil
+	asr.Allocations = append(asr.Allocations, that)
```

Comment:
Perhaps we should nuke this function too?

Reply:
Yeah, good call!